cns/ipampoolmonitor/ipampoolmonitor.go

```diff
@@ -249,3 +248,14 @@
 
 	return nil
 }
+
+//this function sets the values for state in IPAMPoolMonitor Struct
+func (pm *CNSIPAMPoolMonitor) GetStateSnapshot() cns.IpamPoolMonitorStateSnapshot{
+	defer pm.mu.Unlock()
+	pm.mu.Lock()
+
+	return cns.IpamPoolMonitorStateSnapshot {
+		MinimumFreeIps: pm.MinimumFreeIps,
```

Comment:
Can  you also return

 cachedNNC   nnc.NodeNetworkConfig
 updatingIpsNotInUseCount int

Reply:
@neaggarwMS  yes I have added them too to IPAMPoolMonitor in the latest commit, please review if that looks good and the CLI command/end-point we should name for this debug API.